backend/src/main/java/org/polsl/backend/service/HardwareService.java

```diff
@@ -126,4 +127,27 @@ public void editHardware(Long id, HardwareInputDTO request) throws NotFoundExcep
       affiliationHardwareRepository.save(affiliationHardware);
     }
   }
+
+  @Transactional
+  public void deleteHardware(Long id) throws NotFoundException{
+    Hardware hardware = hardwareRepository.findById(id)
+        .orElseThrow(() -> new NotFoundException("sprzęt", "id", id));
+    hardware.setValidTo(LocalDateTime.now());
+    hardwareRepository.save(hardware);
+
+    ComputerSetHardware lastEntryComputerSet = computerSetHardwareRepository.findNewestRowForHardware(id)
+        .orElse(null);
```

Comment:
Słusznie zauważyłeś, że jeśli nie ma żadnego powiązania z inną tabelą, to nie powinniśmy rzucać wyjątku. Jednak tak samo trzeba zrobić niżej, przy wyszukiwaniu połączeń z afiliacjami, oraz wyżej, w edycji.

Reply:
A czy nie miało być tak, że zawsze jest połączenie z Afiliacją? W sensie że nie ma sprzętu nieprzypisanego?

Reply:
Tak, myślę, że ta by było sensownie 😉

Reply:
A racja

Reply:
To chwila XD

Reply:
Zwracam honor, jest dobrze 😄